src/wasm2asm.h

```diff
@@ -161,15 +168,33 @@ class Wasm2AsmBuilder {
     frees[type].push_back(temp);
   }
 
-  IString fromName(Name name) {
+  IString fromName(Name name, NameScope scope) {
```

Comment:
please add a comment for this more-complex function. After a minute or so looking at it I'm still lost...